projects/noloan2/app/src/main/java/hasadna/noloan2/LawsuitPdfActivity.java

```diff
@@ -32,179 +39,243 @@
-  private Date receivedSpamDate = new Date();
-
-  // region Lawsuit form fields
-  // General
-  private String spamType = "הודעה אלקטרונית";
-  private String claimAmount = "סכום תביעה";
+  // Lawsuit optional fields
   private String moreThanFiveLawsuits = "הגיש";
   private String lessThanFiveLawsuits = "לא הגיש/ו";
-  private String claimCaseHaser = "למרות שח\"מ לא נתן את הסכמתו המפורשת מראש לקבלת דבר/י הפרסומת";
+  private String claimCaseHaser =
+      "לאחר שהתובע חזר בו מהסכמתו לקבלת דבר/י פרסומת\nמהנתבע/ים, על-ידי משלוח הודעת סירוב כהגדרתה בחוק";
   private String claimCaseSubscription =
-      "למרות שח\"מ לא נתן את הסכמתו המפורשת מראש לקבלת דבר/י הפרסומת";
-
-  // User
-  private String userPrivateName = "שם פרטי";
-  private String userLastName = "שם משפחה";
-  private String userID = "ת ז משתמש";
-  private String userAddress = "כתובת משתמש";
-  private String userPhone = "טלפון משתמש";
-  private String userFax = "פקס משתמש";
-
-  // First spam company's details
-  private String companyName = "שם החברה";
-  private String companyId = "מספר ח.פ";
-  private String companyAddress = "כתובת החברה";
-  private String companyPhone = "טלפון החברה";
-  private String companyFax = "פקס חברה";
-
-  // Second spam company's details
-  private String company2Name = "שם החברה 2";
-  private String company2Id = "מספר ח.פ 2";
-  private String company2Address = "כתובת החברה 2";
-  private String company2Phone = "טלפון החברה 2";
-  private String company2Fax = "פקס חברה 2";
-  // endregion
+      "למרות שח\"מ לא נתן את הסכמתו המפורשת מראש  \nלקבלת דבר/י הפרסומת";
 
+  EditText receivedDate;
+  DatePickerDialog datePickerDialog;
+  Calendar calendar;
   Button createPdfButton;
 
   @Override
   protected void onCreate(Bundle savedInstanceState) {
     super.onCreate(savedInstanceState);
     setContentView(R.layout.activity_lawsuit_pdf);
-    createPdfButton = (Button) findViewById(R.id.button_createPDF);
 
-    initDirStructure();
+    createPdfButton = findViewById(R.id.button_createPDF);
+    receivedDate = findViewById(R.id.editText_spamDate);
+
     setTimeZones();
+    createPdfButton.setOnClickListener(
+        v -> {
+          checkPermissions();
+          if (permissionGranted) {
+            sharePdf(createPdf());
+          }
+        });
+    receivedDate.setOnClickListener(v -> displayDatePicker());
+  }
 
-    createPdfButton.setOnClickListener(v -> checkPermissionsThenCreatePdf());
+  // Used for date fields in the lawsuit
+  private void setTimeZones() {
+    DATE_TIME_FORMATTER.setTimeZone(TimeZone.getTimeZone(TIME_ZONE));
+    DATE_FORMATTER.setTimeZone(TimeZone.getTimeZone(TIME_ZONE));
   }
 
-  private void createPdf() {
-    // TODO: Success reading license file only from "res/raw". Perhaps should be read from different
-    // dir
-    // Activate iText license
-    LicenseKey.loadLicenseFile(getResources().openRawResource(R.raw.itextkey));
+  // Save PDF to device. Return absFilename of the file
```

Comment:
What is absFilename? Don't use code language in your comments.
I can look at the return line and see that it return "absFilename", I need the comment to understand what it is.

Reply:
This (absFilename), is written not as a name of a specific variable, but as a name convention for `absolute folder + filename` according to this nomenclature: 
https://github.com/hasadna/hasadna/blob/master/NOMENCLATURE.md

I added this comment because from the name of the function you can easily understand what it does, but not exactly what it returns.

I assumed that if we are under the same nomenclature "umbrella" on all repo's projects, it'll be fine to use them as readable language as well. But let me know if to change it :)